Part1_InterruptControllersEvolution.html

```diff
@@ -104,79 +104,78 @@ <h3>PIC</h3>
+	<p>Full documentation of the 8259A can be found <a href="https://pdos.csail.mit.edu/6.828/2005/readings/hardware/8259A.pdf">here</a><p/>.
 
-	<p>The PCI bus has replaced the ISA bus. And the number of devices amount clearly began to 
-	exceed the number 15. Plus instead of the static ISA bus in the PCI bus devices can be added to system dynamically. But luckily, in the PCI bus interrupts can be shared (it means that it is possible to connect many devices to one interrupt line IRQ). In the end, to solve the problem of lack of interrupt lines, it was decided to group interrupts from all of PCI devices to PIRQ lines (Programmable Interrupt Request).<p/>
+	<p>The PCI bus later replaced the ISA bus, and the number of devices began to exceed the number 15. Instead of the static ISA bus, devices in the PCI bus can be added to the system dynamically. In the PCI bus, interrupts can also be shared, since it is possible to connect many devices to one interrupt line IRQ. In the end, to solve the problem of lack of interrupt lines, it was decided to group interrupts from all of the PCI devices to PIRQ lines (Programmable Interrupt Request).<p/>
 
-	<p>For example we have 4 free interrupt lines on the PIC controller and 20 PCI devices. We can combine interrupts by 5 devices at one PIRQx line, and connect these PIRQx lines to the PIC controller. In this case if there is an interrupt on one of PIRQx lines, processor will have to ask all devices connected to this line about the interrupt, to know who is responsible for it, but in the end it solves the problem. The device that connects PCI interrupt lines to PIRQ lines is often called a PIR router.<p/>
+	<p>For example, suppose we have 4 free interrupt lines on the PIC controller and 20 PCI devices. We can combine interrupts from 5 devices into one PIRQx line, and connect these PIRQx lines to the PIC controller. In this case if there is an interrupt on one of PIRQx lines, the processor will have to ask all the devices connected to this line about the interrupt to know who is responsible for it, but in the end it solves the problem. The device that connects PCI interrupt lines to PIRQ lines is often called a PIR router.<p/>
 
-	<p>In this method it is necessary to watch out that PIRQx lines don't connect to lines with ISA interrupts (cause this will produce conflicts) and PIRQx lines are balanced (the more devices we connect to one line, the more devices CPU will need to poll, when it will need to understand which device is responsible for the interrupt).<p/>
+	<p>With this method it is necessary to ensure that PIRQx lines don't connect to lines with ISA interrupts (since this will produce conflicts) and that PIRQx lines are balanced (the more devices we connect to one line, the more devices the CPU will need to poll when it needs to check which device is responsible for the interrupt).<p/>

```

Comment:
What is the difference between "cause" and "since"? Cause I use "cause" all the time when I don't talk about time

Reply:
"cause" is an abbreviation of "because" and is mostly used informally in conversations. "since" is better to use in formal writing.